lib/lualink.c

```diff
@@ -72,6 +77,7 @@ lua_State* Lua_Init(void)
     L = luaL_newstate();
     luaL_openlibs(L);
     Lua_linkctolua(L);
+    lua_sethook(L, hook, LUA_MASKCOUNT, WATCHDOG_FREQ);
```

Comment:
I think this `lua_sethook` is redundant because `Lua_eval` uses `lua_pcall_protected`

Reply:
spot on! initially this was just to setup the hook, but then i ended up having to reset it each time a new external call happened. pushed the fix.